# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/02/07 20:24:13-08:00 davem@nuts.davemloft.net 
#   [SPARC64]: Mask off stack ptr in alloc_user_space() for 32-bit.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# arch/sparc64/kernel/sys_sparc32.c
#   2005/02/07 20:23:33-08:00 davem@nuts.davemloft.net +2 -0
#   [SPARC64]: Mask off stack ptr in alloc_user_space() for 32-bit.
# 
# arch/sparc64/kernel/ioctl32.c
#   2005/02/07 20:23:33-08:00 davem@nuts.davemloft.net +2 -0
#   [SPARC64]: Mask off stack ptr in alloc_user_space() for 32-bit.
# 
diff -Nru a/arch/sparc64/kernel/ioctl32.c b/arch/sparc64/kernel/ioctl32.c
--- a/arch/sparc64/kernel/ioctl32.c	2005-02-09 09:11:38 -08:00
+++ b/arch/sparc64/kernel/ioctl32.c	2005-02-09 09:11:38 -08:00
@@ -562,6 +562,8 @@
 
 	if (!(current->thread.flags & SPARC_FLAG_32BIT))
 		usp += STACK_BIAS;
+	else
+		usp &= 0xffffffffUL;
 
 	return (void *) (usp - len);
 }
diff -Nru a/arch/sparc64/kernel/sys_sparc32.c b/arch/sparc64/kernel/sys_sparc32.c
--- a/arch/sparc64/kernel/sys_sparc32.c	2005-02-09 09:11:38 -08:00
+++ b/arch/sparc64/kernel/sys_sparc32.c	2005-02-09 09:11:38 -08:00
@@ -770,6 +770,8 @@
 
 	if (!(current->thread.flags & SPARC_FLAG_32BIT))
 		usp += STACK_BIAS;
+	else
+		usp &= 0xffffffffUL;
 
 	return (void *) (usp - len);
 }
